http/transport.go

```diff
@@ -201,7 +201,7 @@ func (t *ServerTransport) getListener(network, addr string) (net.Listener, error
 		}
 		ln, ok = pln.(net.Listener)
 		if !ok {
-			return nil, fmt.Errorf("invalid net.Listener")
+			return nil, errors.New("invalid net.Listener")
```

Comment:
maybe wrapping some information:

```suggestion
			return nil, fmt.Errorf("invalid listener type, want net.Listener, got %T", pln)
```